mantidproject/mantid
# Cleanup creating and passing boost::shared_ptr.

Description of work.

This updates several places I found while working on #16046 where boost::shared_ptr is constructed from a raw pointer instead of `boost::make_shared`. In the `extractMetadata` member function I found that ownership wasn't being transferred and therefore passing a reference makes more sense.

**To test:**



Review changes and verify that the builds pass.

This is a small change with no issue number

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation? --> N/A
- [ ] If there are changes in the release notes then do they describe the changes appropriately? -->N/A
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed? --?N/A
- [ ] How do the changes handle unexpected situations, e.g. bad input? -->N/A
- [ ] Has the relevant documentation been added/updated? --> N/A
- [ ] Is user-facing documentation written in a user-friendly manner? --> N/A
- [ ] Has developer documentation been updated if required? --> N/A
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
